### PR TITLE
UBI8 on OpenShift fixes

### PR DESCRIPTION
**Pull request by KyWa**
New:
- BuildConfig Template for UBI

Updated:
- Dockerfile.ubi8 contains fixes for running UBI on OpenShift
- entrypoint.sh is also included in this to ensure proper NSS Wrapper
  for RHEL based images
- deploy-code-server-no-webdav-template.yml was targeting a UBI image
  but had a "dockerfile" showing apt-get commands

The Dockerfile.ubi8 has been tested on OpenShift 4.6 being built locally
and run via the deploy Template.

### CODE REVIEW

File: container/base/bin/entrypoint.sh [hunk 1]
```diff
@@ -65,13 +64,15 @@ if [ -f /usr/local/bin/npm-setup.sh ]; then
...
 # kludge: opinionated defaults
-#if [ ! -e ${HOME}/.local/share/code-server/User/settings.json ]; then
-#mkdir -p ${HOME}/.local/share/code-server/User
-#echo "{
-#    "workbench.colorTheme": "Abyss",
-#    "terminal.integrated.defaultProfile.linux": "bash",
-#    "telemetry.enableTelemetry": false
-#}" > ${HOME}/.local/share/code-server/User/settings.json
-#fi
+# the below is a requirement for UBI based images (and wont break antything Debian based)
```
Comment (codekow) [inline container/base/bin/entrypoint.sh:67]: antything = anything

File: container/base/bin/entrypoint.sh [hunk 2]
```diff
@@ -65,13 +64,15 @@ if [ -f /usr/local/bin/npm-setup.sh ]; then
...
+if [ ! -e ${HOME}/.local/share/code-server/User/settings.json ]; then
```
Comment (codekow) [inline container/base/bin/entrypoint.sh:68]: The code below was causing some issues when I was testing... I'll test again, but those were just my opinionated settings... they can be removed :)

Comment (KyWa) [reply container/base/bin/entrypoint.sh:68]: The only thing that really needed to be added to make it work on OCP was the `"terminal.integrated.shell.linux": "/bin/bash",`. OCP with its random UID (even with the NSS wrapper and related ENV vars) wasn't being respected by `code-server` and was still seeing that it had its shell set to `/sbin/nologin`. This config (which apparently changes soon in Code Server to a "profile" approach) gets around that. So its possible the NSS wrapper may not even be required and can be cleaned up a little bit. I'll test that later, but for now this is a known working config (yay).

File: container/base/bin/entrypoint.sh (hunk 1 above)
Comment (KyWa) [reply container/base/bin/entrypoint.sh]: Fixed. I promise I know how to spell.

File: container/base/bin/entrypoint.sh (hunk 2 above)
Comment (codekow) [reply container/base/bin/entrypoint.sh:68]: Sounds good.

**Review by codekow — APPROVED** [review]
Looks good. I'll approve it.

**Pull request merged**